tests/test_pytest_arraydiff.py

```diff
@@ -176,3 +176,54 @@ def test_single_reference(self, spam):
 
 def test_nofile():
     pass
+
+class BaseTestClass:
+    arrays = None
+    @pytest.mark.array_compare(reference_dir=reference_dir, file_format='text', derive_classes=True)
+    def test_array_one(self):
+        return self.array
+class TestDerivedOne(BaseTestClass):
+    array = np.arange(3 * 4).reshape((3, 4))
+    
+class TestDerivedTwo(BaseTestClass):
+    array = np.arange(2 * 5).reshape((2, 5))
+
+
+
+DERIVED_FAILING = """
+import pytest
+import numpy as np
+class BaseTestClass:
+    arrays = None
+    @pytest.mark.array_compare(reference_dir="{reference_dir}", file_format='text')
+    def test_array_one(self):
+        return self.array
+class TestDerivedOne(BaseTestClass):
+    array = np.arange(3 * 4).reshape((3, 4))
+    
+class TestDerivedTwo(BaseTestClass):
+    array = np.arange(2 * 5).reshape((2, 5))
+"""
+
+
+def test_derived_fails():
+
+    tmpdir = tempfile.mkdtemp()
```

Comment:
Can probably just use `tmp_path` fixture? https://docs.pytest.org/en/stable/how-to/tmp_path.html